module/proto/peggy/v1/confirm.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+package peggy.v1;
+
+option go_package = "github.com/althea-net/peggy/module/x/peggy/types";
+
+// ConfirmType is the cosmos type of a confirmation of a given object from the eth signers
+enum ConfirmType {
+  CONFIRM_TYPE_UNSPECIFIED = 0;
+  CONFIRM_TYPE_VALSET      = 1;
+  CONFIRM_TYPE_BATCH       = 2;
+  CONFIRM_TYPE_LOGIC       = 3;
+}
+
+// ConfirmLogicCall
+// When validators observe a MsgRequestBatch they form a batch by ordering
+// transactions currently in the txqueue in order of highest to lowest fee,
+// cutting off when the batch either reaches a hardcoded maximum size (to be
+// decided, probably around 100) or when transactions stop being profitable
+// (TODO determine this without nondeterminism) This message includes the batch
+// as well as an Ethereum signature over this batch by the validator
+// -------------
+message ConfirmLogicCall {
+  string invalidation_id    = 1;
+  uint64 invalidation_nonce = 2;
+  string eth_signer         = 3;
+  string orchestrator       = 4;
+  string signature          = 5;
+}
+
+// ConfirmBatch
+// When validators observe a MsgRequestBatch they form a batch by ordering
+// transactions currently in the txqueue in order of highest to lowest fee,
+// cutting off when the batch either reaches a hardcoded maximum size (to be
+// decided, probably around 100) or when transactions stop being profitable
+// (TODO determine this without nondeterminism) This message includes the batch
+// as well as an Ethereum signature over this batch by the validator
+// -------------
+message ConfirmBatch {
+  uint64 nonce          = 1;
+  string token_contract = 2;
+  string eth_signer     = 3;
+  string orchestrator   = 4;
+  string signature      = 5;
+}
+
+// ValsetConfirm
+// this is the message sent by the validators when they wish to submit their
+// signatures over the validator set at a given block height. A validator must
```

Comment:
validators on cosmos? I don't understand the "submit signatures over the validator set". Why do you need to request and confirm the validator set in the first place?

Reply:
This is because we need to keep the owners of the contract on ETH the same as the valset on cosmos. Each cosmos validator has an eth key associated and that makes up the multisig

Reply:
I understand that but why don't we submit a single `UpdateValset` tx to ethereum with the keys and updated power instead? The request + confirm design is something that I really don't get

Reply:
Requests are coordinating points for the external signers to work with.

If all the orchestrators tried to sign the 'current validator set' they would get different values since they are operating outside of the tendermint consensus code. For validator sets we've automated validator set request creation, but we can't remove it without moving the Ethereum signing into tendermint iself. Which maybe possible in the future from what I've heard.

If we could import an ethereum private key from the validators key store and perform signatures in the module then requests could be eliminated as a concept. I think this should be a long term goal.

For now we're stuck with them.